exampleLitActions/src/fetch.ts

```diff
@@ -0,0 +1,129 @@
+  if (!value) {
+    console.warn(`Environment variable ${name} is not set`);
+  }
+  return value || "";
+};
+
+// Environment variables
+const ETHEREUM_PRIVATE_KEY = getEnv("ETHEREUM_PRIVATE_KEY");
+
+// Define the Lit Action code for conditional signing
+const litActionCode = `
+async () => {  
+  const url = "https://api.weather.gov/gridpoints/TOP/31,80/forecast";
+  const resp = await fetch(url).then((response) => response.json());
+  const temp = resp.properties.periods[0].temperature;
+  // only sign if the temperature is above 60. if it's below 60, exit.
+  if (temp < 60) {
+    return;
+  }
+  
+  // this requests a signature share from the Lit Node
+  // the signature share will be automatically returned in the HTTP response from the node
+  // all the params (toSign, publicKey, sigName) are passed in from the LitJsSdk.executeJs() function
+  const sigShare = await LitActions.signEcdsa({ toSign, publicKey, sigName });
+};
+`;
+
+// Define the fetchResult type with success and error properties
+interface fetchResult {
+  response: any;
+  success: boolean;
+  error?: any;
+}
+
+/**
+ * Execute fetch inside a Lit Action
+ * @returns Result of the fetch
+ */
+export const fetch = async (): Promise<fetchResult> => {
+  try {
+    // Create ethers wallet for signing
+    const ethersWallet = new ethers.Wallet(
+      ETHEREUM_PRIVATE_KEY,
+      new ethers.providers.JsonRpcProvider(LIT_RPC.CHRONICLE_YELLOWSTONE)
+    );
+    
+    const message = new Uint8Array(
+      await crypto.subtle.digest('SHA-256', new TextEncoder().encode('Hello world'))
+    );
+    
+    const litNodeClient = new LitNodeClient({
+      alertWhenUnauthorized: false,
+      litNetwork: "datil-dev",
+      debug: true,
+    });
+    
+    await litNodeClient.connect();
+    
+    const sessionSigs = await litNodeClient.getSessionSigs({
+      chain: "ethereum",
+      expiration: new Date(Date.now() + 1000 * 60 * 10).toISOString(), // 10 minutes
+      resourceAbilityRequests: [
+        {
+          resource: new LitActionResource("*"),
+          ability: LIT_ABILITY.LitActionExecution,
+        },
+      ],
+      authNeededCallback: async ({
+        uri,
+        expiration,
+        resourceAbilityRequests,
+      }) => {
+        const toSign = await createSiweMessage({
+          uri,
+          expiration,
+          resources: resourceAbilityRequests,
+          walletAddress: await ethersWallet.getAddress(),
+          nonce: await litNodeClient.getLatestBlockhash(),
+          litNodeClient,
+        });
+        return await generateAuthSig({
+          signer: ethersWallet,
+          toSign,
+        });
+      },
+    });
+    
+    const signatures = await litNodeClient.executeJs({
+      code: litActionCode,
+      sessionSigs,
+      // all jsParams can be used anywhere in your litActionCode
+      jsParams: {
+        toSign: message,
+        publicKey:
+          "0x02e5896d70c1bc4b4844458748fe0f936c7919d7968341e391fb6d82c258192e64",
```

Comment:
This should not be hardcoded, we should either use a PKP given to us as an ENV, or we should mint a new PKP like we do in `exampleLitActions/src/conditionalSigning.ts`